src/ol/expr/expression.js

```diff
@@ -1051,6 +1059,19 @@ function parsePaletteArgs(encoded, context) {
   return [index, ...parsedColors];
 }
 
+/**
+ * @type ArgValidator
+ */
+function parseToNumberArgs(encoded, context, parsedArgs) {
+  const operation = encoded[0];
+  if (isNaN(Number(encoded[1]))) {
+    throw new Error(
+      `The passed argument for the ${operation} operation, resulted in NaN`
+    );
+  }
```

Comment:
Here you are checking the actual type of the encoded argument, so if it's an expression like `['get', 'id']` it will fail at compilation time.

You shouldn't need this new function, see my comment above.